### PR TITLE
Add support for safetensors when reading checkpoints

### PR DESCRIPTION
**Pull request by tjs-intel**
# What does this PR do?

The [TinyLlama model](https://huggingface.co/TinyLlama/TinyLlama-1.1B-Chat-v1.0) only has checkpoints in the form of `model.safetensors`. This checkpoint needs to be included in the list of checkpoints that is passed to DeepSpeed in order for the model to function properly when initialized with DeepSpeed.

This PR adds the safetensor checkpoints to the list of checkpoints passed to DeepSpeed.

**Note**: This change requires an [upstream commit in microsoft/DeepSpeed](https://github.com/microsoft/DeepSpeed/commit/c8c57b8c24ad761267ab43b0252fb62f0c30d289) to be merged downstream to HabanaAI/DeepSpeed in order for DeepSpeed to support the provided safetensor format.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by tjs-intel** [comment]
PR applying patch to support safetensors to HabanaAI/DeepSpeed: https://github.com/HabanaAI/DeepSpeed/pull/2

**Comment by tjs-intel** [comment]
@dvarshney-habana please assign reviewers as appropriate

**Comment by vivekgoe** [comment]
@tjs-intel @dvarshney-habana this looks same as https://github.com/HabanaAI/optimum-habana-fork/pull/15 . Please confirm, if its same then other one is already in queue to merge.

File: optimum/habana/checkpoint_utils.py [hunk 1]
```diff
@@ -58,7 +58,9 @@ def get_checkpoint_files(model_name_or_path, local_rank, token=None):
 
     # Extensions: .bin | .pt
     # Creates a list of paths from all downloaded files in cache dir
-    file_list = [str(entry) for entry in Path(cached_repo_dir).rglob("*.[bp][it][n]") if entry.is_file()]
+    path = Path(cached_repo_dir)
+    globs = ["*.bin", "*.pt", "*.safetensors"]
```
Comment (puneeshkhanna) [inline optimum/habana/checkpoint_utils.py:62]: Don't we need some kind of if else check ; cached repo dir can have both .safetensors and .pt files.

**Comment by tjs-intel** [comment]
@vivekgoe I will confirm on Monday that #15 resolves my issue

File: optimum/habana/checkpoint_utils.py (hunk 1 above)
Comment (tjs-intel) [reply optimum/habana/checkpoint_utils.py:62]: The code that is already there loads `.bin` and `.pt` files, can the cached repo have both of those as well? I might be misunderstanding something

**Comment by tjs-intel** [comment]
Resolved with #15 which is already good to merge

File: optimum/habana/checkpoint_utils.py (hunk 1 above)
Comment (puneeshkhanna) [reply optimum/habana/checkpoint_utils.py:62]: Yes it can have both.

**Pull request closed without merging**